contrib/parseq-examples/src/main/java/com/linkedin/parseq/example/batching/BatchingClientExample.java

```diff
@@ -0,0 +1,70 @@
+/* $Id$ */
+package com.linkedin.parseq.example.batching;
+
+import java.util.List;
+
+import com.linkedin.parseq.Engine;
+import com.linkedin.parseq.Task;
+import com.linkedin.parseq.batching.BatchEntry;
+import com.linkedin.parseq.batching.BatchingStrategy;
+import com.linkedin.parseq.batching.BatchingSupport;
+import com.linkedin.parseq.example.common.AbstractExample;
+import com.linkedin.parseq.example.common.ExampleUtil;
+import com.linkedin.parseq.promise.Promises;
+
+
+/**
+ * @author Jaroslaw Odzga (jodzga@linkedin.com)
+ */
+public class BatchingClientExample extends AbstractExample {
+
+  public static class ExampleBatchingStrategy extends BatchingStrategy<Long, Long> {
+
+    @Override
+    public Long groupForKey(Long key) {
+      //lets say all tasks can be grouped into one batch
+      return 0L;
+    }
+
+    @Override
+    public Task<?> taskForBatch(final Long group, final List<? extends BatchEntry<Long, Object>> batch) {
```

Comment:
Is there a way to not lose the type information here (Object)? Seems like in the case of rest.li you'd end up casting these to Get requests or BatchGet requests  all the time.